### PR TITLE
kola/tests/ignition: add test for versioned s3 objects

### LINKED ISSUE #933: Add test for Ignition PR 643
https://github.com/coreos/ignition/pull/643 needs a test that needs to be in kola, not the bb tests.

### PR DESCRIPTION
**Pull request by arithx**
Adds a test for pulling versioned s3 objects (a new feature in
Ignition).

Tests both a null versionId (AWS uses this for the original version of a
file) and a later versionId.

Closes #933

### CODE REVIEW

**Comment by arithx** [comment]
This PR needs to be updated once coreos/ignition#643 merges and propogates to the relevant distributions / channels to have the `MinVersion` set.

**Comment by lucab** [comment]
:heart:

File: kola/tests/ignition/resource.go [hunk 1]
```diff
@@ -147,6 +147,40 @@ func init() {
...
+			      "contents": {
+				  "source": "http://s3-us-west-2.amazonaws.com/kola-fixtures/resources/versioned?versionId=null"
+			      },
+			      "mode": 420
+			  },
+			  {
+			      "filesystem": "root",
+			      "path": "/var/resource/latest",
+			      "contents": {
+				  "source": "http://s3-us-west-2.amazonaws.com/kola-fixtures/resources/versioned?versionId=RDWqxfnlcJOSDf1.5jy6ZP.oK9Bt7_Id"
```
Comment (lucab) [inline kola/tests/ignition/resource.go]: Do we have content in this resource that changed across versions and that we can check once the node is provisioned?

Comment (arithx) [reply kola/tests/ignition/resource.go]: Yes, in the original version (where `versionId` is `null`), the contents of the file are `original`, while in this version the contents are `updated`, see lines 245-248.

**Comment by lucab** [comment]
@arithx I think we can update this `DNM` now. This feature was released in Ignition `0.29.0` and reached CL Alpha in 1995.0.0.

**Comment by arithx** [comment]
@lucab yup, `0.29.1` is also in RPM form, merging now.

**Comment by bgilbert** [comment]
Ignition 0.29.x isn't in beta or stable yet, so this will need a `MinVersion`.

**Pull request merged**